### PR TITLE
Fix Archlinux unsupported os during Vulnerability Detector scan.

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#12429|

## Description

As a consequence of the changes of https://github.com/wazuh/wazuh/pull/11207 the `os_major` information can't be parsed from `os_version` when it arrives at the manager and it is set as NULL in the global database for rolling distributions. Due to the case previously mentioned vulnerability detector skips the scan. 

## Scan-build report

image

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report

### CODE REVIEW

**Comment by MiguelazoDS** [comment]
# DoD

## Agent info

### Debian 

image

### Arch Linux

image

## Vulnerability detector scan

### Log

image

### Vulnerabilities found

image

File: src/shared/remoted_op.c [hunk 1]
```diff
@@ -108,6 +107,8 @@ void parse_uname_string (char *uname,
                     match_size = match[1].rm_eo - match[1].rm_so;
                     os_malloc(match_size +1, osd->os_major);
                     snprintf(osd->os_major, match_size + 1, "%.*s", match_size, osd->os_version + match[1].rm_so);
+                } else {
+                    os_strdup("", osd->os_major);
```
Comment (DProvinciani) [inline src/shared/remoted_op.c:111]: As this is now a valid case, I would add a unit test case to cover it.

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Apply changes according to the comment.

File: src/shared/remoted_op.c (hunk 1 above)
Comment (MiguelazoDS) [reply src/shared/remoted_op.c:111]: Added
image

Comment (MiguelazoDS) [reply src/shared/remoted_op.c:111]: Added
image

Comment (MiguelazoDS) [reply src/shared/remoted_op.c:111]: Added
image

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Review by vikman90 — CHANGES_REQUESTED** [review]
Hi @MiguelazoDS,

Vulnerability Detector now runs the scan on Arch Linux agents, but it prints the following error:
```
(5568): Invalid OS information of the agent '001'
```

I'm afraid that Vuln Detector gets a null value from the column `os_major` at the agent's DB table `sys_osinfo`.

Please take a look at this prior to merging this PR.

Best.

**Comment by MiguelazoDS** [comment]
Hi @vikman90, 

There's another approach in the latest commit. I didn't take into account the value in `sys_osinfo` table. Anyway, we still have `os_version` as an empty string in the global database and NULL in `sys_osinfo`, but the `os_major` hardcoded value was reverted.
The evidence was added to https://github.com/wazuh/wazuh/pull/12449#issuecomment-1050117955 and there's no error, thanks for the review. On the other hand, that function is huge and is not tested.

**Review by vikman90 — APPROVED** [review]

**Pull request merged**